src/worker/generate-kmp.ts

```diff
@@ -0,0 +1,38 @@
+import { generateKmpJson } from "./generate-kmp-json";
+import { createZipWithFiles } from "./generate-zip";
+import {
+  compileModelFromLexicalModelSource,
+  WordListFromArray,
+} from "@predictive-text-studio/lexical-model-compiler";
+
+/**
+ * Generate kmp.json and model.js file, then create a kmp file for the dictionary
+ *
+ * Format of the lexical model file: {model_id}.model.js
+ * the compiled code generated by the @predictive-text-studio/lexical-model-compiler
+ *
+ * @param langName
+ * @param bcp47Tag
+ * @param sources
+ * @param modelId
+ */
+export async function generateKmp(
+  langName: string,
+  bcp47Tag: string,
+  sources: WordListFromArray[],
+  modelId: string
+): Promise<ArrayBuffer> {
+  const kmpJsonFile = generateKmpJson({
+    languages: [{ name: langName, id: bcp47Tag }],
```

Comment:
@Xuechunqiu a thing we didn't notice earlier is that the `kmp.json` file needs to have the same modeld as in the `.kmp` file; however, since we didn't provide it as an argument here, the name in the `kmp.json` may not match the filename here.

Since `generateKmpJson()` automatically generates its own model ID, this **will** get out of sync, and will not install in Keyman (I tried it!). The real solution is to change `generateKmpJson()` so that it accepts a model ID and does not generate it itself, however, this will make it work in the meantime:

```suggestion
    languages: [{ name: langName, id: bcp47Tag }],
    // TODO: change generateKmpJson() to accept a modelID argument instead of splitting it here:
    authorID: modelID.split(".")[0],
    modelTag: modelId.split(".")[2],
```

Reply:
I see what you mean. In addition to these quick changes, do I need to change generateKmpJson() for now, or do it later?

Reply:
I think that's best kept for a seperate PR :)

Reply:
Update: it's issue #91 if you want to tackle it later